v0/destinations/pinterestConversion/transform.js

```diff
@@ -0,0 +1,266 @@
+const _ = require("lodash");
+const { defaultPostRequestConfig } = require("../../util");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  defaultRequestConfig,
+  isDefined,
+  isDefinedAndNotNull,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  constructPayload,
+  defaultBatchRequestConfig
+} = require("../../util");
+const {
+  processUserPayload,
+  processCommonPayload,
+  deduceEventName,
+  setIdPriceQuantity,
+  checkUserPayloadValidity
+} = require("./utils");
+
+const { MAX_BATCH_SIZE, USER_CONFIGS, CUSTOM_CONFIGS } = require("./config");
+
+const responseBuilderSimple = finalPayload => {
+  const response = defaultRequestConfig();
+  response.endpoint = "https://ct.pinterest.com/events/v3";
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = finalPayload;
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+};
+
+const identifyPageResponseBuilder = (message, { Config }) => {
+  const { appId, advertiserId } = Config;
+  if (!isDefined(advertiserId)) {
+    throw new CustomError(
+      "Advertiser ID is mandatory for sending events to Pinterest Conversion API",
+      400
+    );
+  }
```

Comment:
We are already checking it in the first place so we can remove it from here.

Reply:
done